bool detection_allowed
float32 layer